src/rownode.h

```diff
@@ -15,30 +15,31 @@ struct DownloadInfo
     bool paused;
 };
 
-class RowNode : public Node
+class RowNode : public Node, std::enable_shared_from_this<RowNode>
```

Comment:
I think it would be better to have `Node` itself inheriting from `std::enable_shared_from_this`